test/model/message_list_test.dart

```diff
@@ -1704,6 +1704,25 @@ void main() {
       }
     });
   });
+  group('topics compared case-insensitively', () {
+    void doTest(String description, String topicA, String topicB, bool expected) {
+      test(description, () {
+        final stream = eg.stream();
+        final messageA = eg.streamMessage(stream: stream, topic: topicA);
+        final messageB = eg.streamMessage(stream: stream, topic: topicB);
+        check(haveSameRecipient(messageA, messageB)).equals(expected);
+      });
+    }
+
+    doTest('same case, all lower',               'abc',  'abc',  true);
+    doTest('same case, all upper',               'ABC',  'ABC',  true);
+    doTest('same case, mixed',                   'AbC',  'AbC',  true);
+    doTest('same non-letter chars',              '嗎',    '嗎',    true);
+    doTest('different case',                     'aBc',  'ABC',  true);
+    doTest('different case, same diacritics',    'AbÇ',  'aBç',  true);
+    doTest('same letters, different diacritics', 'ma',   'mǎ',   false);
+    doTest('having different non-letter chars',  'ma 嗎', 'mǎ 馬', false);
```

Comment:
```suggestion
    doTest('same letters, different diacritics', 'ma',   'mǎ',   false);
    doTest('having different non-letter chars',  '嗎',    '馬',    false);
```

(plus the renaming above)

Otherwise the last test doesn't really exercise the comparison of the two sinographs — the two names are already going to be different because of the "a" vs "ǎ".